packages/remix-dev/vite/plugin.ts

```diff
@@ -737,6 +747,88 @@ export const remixVitePlugin: RemixVitePlugin = (options = {}) => {
           }
         };
       },
+      writeBundle: {
+        // After the SSR build is finished, we inspect the Vite manifest for
+        // the SSR build and move all server assets to client assets directory
+        async handler() {
+          if (!ssrBuildContext.isSsrBuild) {
+            return;
+          }
+
+          invariant(
+            cachedPluginConfig,
+            "Expected plugin config to be cached when writeBundle hook is called"
+          );
+
+          invariant(
+            resolvedViteConfig,
+            "Expected resolvedViteConfig to exist when writeBundle hook is called"
+          );
+
+          let { assetsBuildDirectory, serverBuildPath, rootDirectory } =
+            cachedPluginConfig;
+          let serverBuildDir = path.dirname(serverBuildPath);
+
+          let ssrViteManifest = await loadViteManifest(serverBuildDir);
+          let clientViteManifest = await loadViteManifest(assetsBuildDirectory);
+
+          let clientAssetPaths = new Set(
+            Object.values(clientViteManifest).flatMap(
+              (chunk) => chunk.assets ?? []
+            )
+          );
+
+          let ssrOnlyAssetPaths = new Set(
+            Object.values(ssrViteManifest)
+              .flatMap((chunk) => chunk.assets ?? [])
+              // Only move assets that aren't in the client build
+              .filter((asset) => !clientAssetPaths.has(asset))
+          );
+
+          let movedAssetPaths = await Promise.all(
+            Array.from(ssrOnlyAssetPaths).map(async (ssrAssetPath) => {
+              let src = path.join(serverBuildDir, ssrAssetPath);
+              let dest = path.join(assetsBuildDirectory, ssrAssetPath);
+              await fse.move(src, dest);
+              return dest;
+            })
+          );
+
+          let logger = vite.createLogger();
```

Comment:
I'm not sure if it matters much, but there is `logger` from `resolvedViteConfig` and I used that before.

```suggestion
          let logger = resolvedViteConfig.logger;
```